tests/Data/Text/AhoCorasickSpec.hs

```diff
@@ -398,6 +399,23 @@ spec = parallel $ do
       in
         Replacer.run replacer haystack `shouldBe` expected
 
+  describe "Searcher.containsAll" $ do
+
+    prop "never reports true for empty needles" $ \ (haystack :: Text) ->
+      Searcher.containsAll CaseSensitive [""] haystack `shouldBe` False
+
+    prop "is equivalent to sequential Text.isInfixOf calls" $ \ (needles :: [Text]) (haystack :: Text) ->
+      not (any Text.null needles) ==>
```

Comment:
It could make sense to use a custom generator for generating non-empty needles (maybe a `newtype NonEmptyText`).